content/blog/open-sourcing-packetrusher-a-5g-core-performance-tester.md

```diff
@@ -0,0 +1,67 @@
+disable: false
+tags:
+  - opensource
+  - 5g
+  - gnodeb
+  - ue
+  - simulator
+---
+In the fast-evolving landscape of 5G technology, the demand for robust and efficient testing tools has never been higher. Enter PacketRusher, a cutting-edge 5G Core Network performance testing tool. As the lead developer behind this groundbreaking project, I am thrilled to share the power and potential PacketRusher holds in revolutionizing the way we test and optimize 5G networks.
+
+## PacketRusher
+
+PacketRusher is a tool dedicated to the performance testing and automatic validation of 5G Core Networks. It tests a 5G Core Network via its external interfaces, as defined by the 3GPP : N1, N2 and N3 interfaces. The tool can emulate various types of User Equipment (eg. 5G phones or 5G-enabled IoT devices) and gNodeBs (eg. antennas in Radio Access Network), and it can validate and load-tests the 5G core network’s Control Plane functions using a set of pre-defined procedures on top of both New Generation Application Protocol (NGAP) and Non-access Stratum (NAS-5GS) 5G protocols. It enables users to tests 5G Core Networks with up to 100k simulated UEs simultaneously all from a simple Linux virtual machine, without the needs for expensive commercial tools, or 100k phones on a table ;-)
+
+## Architecture
+Before we understand how PacketRusher works, we need to understand the components of a minimal 5G network:
+
+![High-level diagram of a 5G deployment](/img/high-level-diagram-of-a-5g-deployment.png "High-level diagram of a 5G deployment")
+
+To summarize, there are essentially three main components:
+- User Equipment, which is any 5G device used directly by a user to communicate. For example, a smartphone or an IoT device.
+- The gNodeB, which is the radio used for wireless communication between the User Equipment and the 5G network core.
+- And finally, the 5G network core itself, which manages all mobile network functionalities. This is the component that will manage phone authentication in the network, quality of service, routing to the data network, and so on...
+
+
+![﻿High-level diagram of the PacketRusher's architecture and its interaction with a 5G Core's AMF and UPF](/img/packetrusher-architecture.png "﻿High-level diagram of the PacketRusher's architecture and its interaction with a 5G Core's AMF and UPF")
+
+PacketRusher's behavior is quite simple; it essentially simulates two of the three major components of a 5G network: 
+- the User Equipments, i.e. the phones
+- and the gNodeB, i.e. the antennas.
+- At the same time, it connects to the 5G Core via its external interfaces, as if it were a black box.
+PacketRusher simulates both components en masse, which means it can simulate several user devices and gNodeBs at the same time.
+
+For testing, PacketRusher implements main control plane procedures:
+  * Supports UE attach/detach (registration/authentifcation/security mode) procedures
+  * Supports Create/Delete PDU Sessions,  up to 15 PDU Sessions per UE
```

Comment:
Here it should be "Supports create/delete PDU sessions, up to 15 PDU sessions per EU

By the way, have we defined PDU yet? If not, it makes sense to define it in its first instance of use by spelling it out completely and then writing (PDU) after it. For example, Primary Digital Unit (PDU)... (I know that's probably not what it means, but I don't know what it means... hence my suggestion :-)